File: towerofhanoi.cpp
#include<bits/stdc++.h>
using namespace std;

void towerOfHanoi(int disk,char x,char y,char z){
    if(disk==1){
        cout<<"Move disk from"<<" "<<x<<" "<<"to"<<" "<<y<<endl;
        return;
    }
    towerOfHanoi(disk-1,x,z,y);
    cout<<"Move disk from"<<" "<<x<<" "<<"to"<<" "<<y<<endl;
    towerOfHanoi(disk-1,z,y,x);
}

int main(){
    int disc;
    cout<<"Enter the number of discs: ";
    cin>>disc;
    towerOfHanoi(disc,'A','C','B');
    return 0;
}

File: Tower_Of_Hanoi.py
def TOH(number, s, d, a):
    if number == 1:
        print("Move disk 1 from source", s, "to destination", d)
        return
    TOH(number - 1, s, a, d)
    print("Move disk", number, "from source", s, "to destination", d)
    TOH(number - 1, a, d, s)


n = 3
TOH(n, 'A', 'B', 'C')
